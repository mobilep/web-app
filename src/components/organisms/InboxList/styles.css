.Inbox-empty-state {
	width: 70%;
	font-size: 14px;
	color: var(--grey3);
	text-align: center;
	margin: 0 auto;
}

.Inbox-search-wrapper {
	width: 100%;
	padding: 0 20px 0 30px;
	margin-bottom: 30px;
	height: 34px;
}

.InboxList-list {
	overflow-y: auto;
	margin-bottom: 0;
}

.InboxList {
	display: flex;
	flex-direction: column;
	max-height: 100%;
	height: 100%;
	flex: 1;
}

@media (max-width: 768px) {
	.Inbox-search-wrapper {
		width: 100%;
		padding: 0 10px;
	}

	.InboxList-list {
		flex: 1;
		width: 100%;
		overflow-x: hidden;
	}

	.InboxList-list .UserCardList-item {
		padding-left: 10px;
	}
}
