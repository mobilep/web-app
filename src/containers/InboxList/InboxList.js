import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import { InboxList } from '../../components';
import { isChatType } from '../../helpers/inboxListHelper';
import { chatType } from '../../constants/common';

const mapStateToProps = ({ inboxInfo: { fetching, inboxList } }, props) => ({
	fetching,
	inboxList: inboxList.filter(isChatType(chatType.INBOX)),
	...props,
});

export default withRouter(connect(mapStateToProps)(InboxList));
