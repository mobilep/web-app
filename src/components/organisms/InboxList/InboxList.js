import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { LanguageContext } from '../../../utils';
import { SearchBox, UserCardList } from '../..';
import { CARD_TYPES } from '../../molecules/UserCard/constants';

import './styles.css';
import { matchId } from '../../../helpers/arrayHelpers';

const { Consumer: LanguageConsumer } = LanguageContext;

class InboxList extends Component {

	state = {
		inboxList: [],
		search: false,
		searchTerm: '',
	}

	static getDerivedStateFromProps (props) {
		const { inboxList } = props;

		return inboxList ? { inboxList } : null;
	}

	componentDidUpdate (prevProps) {
		if (prevProps.match.params.view !== this.props.match.params.view) {
			// eslint-disable-next-line react/no-did-update-set-state
			this.setState({ searchTerm: '' });
		}
	}

	getFilteredList (list, termValue) {
		if (!termValue) return list;

		const searchTerm = termValue.trim();
		return list.filter((item) => {
			if (item._recipient) {
				return this.isChatNameMatchSearchTerm(searchTerm)(item._recipient.name);
			}

			return this.isChatNameMatchSearchTerm(searchTerm)(item.title || '');
		});
	}

	isChatNameMatchSearchTerm = (searchTerm) => (chatName) => {
		return chatName.toLowerCase().includes(searchTerm);
	}

	handleSearchBoxChange = (term) => {
		const searchTerm = term.toLowerCase();
		this.setState({ search: true, searchTerm });
	};

	handleCardSelect = (inboxId) => {
		const { history, inboxList } = this.props;
		const inbox = inboxList.find(matchId(inboxId));
		const view = inbox.users ? 'group-chat' : 'chat';
		return history.push(`/inbox/${view}/${inboxId}`);
	}

	render () {
		const {
			activeCard,
			fetching,
			match: { params },
		} = this.props;
		const { inboxList, search, searchTerm } = this.state;

		const listData = this.getFilteredList(inboxList, searchTerm);
		const listType = CARD_TYPES.CONVERSATION_CARD;

		if (fetching) return null;
		return (
			<LanguageConsumer>
				{({ inbox }) => (
					<div className='InboxList'>
						<div className='Inbox-search-wrapper'>
							<SearchBox
								name='searchbox'
								category={''}
								onChange={this.handleSearchBoxChange}
								view={params.view}
								placeholder={inbox.searchLabel}
							/>
						</div>
						{listData.length ? (
							<UserCardList
								className='InboxList-list'
								activeCard={activeCard}
								listData={listData}
								inboxList={this.props.inboxList}
								onSelectCard={this.handleCardSelect}
								type={listType}
							/>
						) : (
							<div className='Inbox-empty-state'>
								<span>{search ? inbox.noResults : inbox.emptyState}</span>
							</div>
						)}
					</div>
				)}
			</LanguageConsumer>
		);
	}
}

InboxList.propTypes = {
	activeCard: PropTypes.string,
	fetching: PropTypes.bool,
	history: PropTypes.object,
	inboxList: PropTypes.array,
	match: PropTypes.object,
};

InboxList.defaultProps = {
	onSelectCard: () => {},
};

export default InboxList;
