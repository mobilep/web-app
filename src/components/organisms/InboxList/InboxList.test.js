import React from 'react';
import { shallow } from 'enzyme';

import InboxList from './InboxList';

describe('Organism - InboxList', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<InboxList match={{ params: { view: 'chat' } }} />
		);
		expect(wrapper).toMatchSnapshot();
	});

});

