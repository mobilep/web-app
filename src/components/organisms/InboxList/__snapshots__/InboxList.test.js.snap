// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organism - InboxList Should render 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
